### DISCUSSION TITLE
How to apply a element condition via plugin as a default rule?

### DISCUSSION THREAD

**Discussion by vnali** [craftcms/cms#11890] [Q&A] [answered]
I am trying to figure out how to automatically apply an element condition rule to an element type index page via a plugin.
I was able to write a condition rule type, but it seems all kinds of rule types need to be selected by the end-user as a filter. Still, I need that filter to be applied automatically in this particular case.
My question is can I have a rule which applies to the element index page automatically? Something like a default filter which end user has no control over it?

**Comment by brandonkelly** [comment, marked as answer]
If you don’t need users to be able to edit the condition, you don’t really need a condition in the first place. Instead you can modify the element criteria for your element sources, by adding an event handler to the element `EVENT_REGISTER_SOURCES` event.

```php
use craft\elements\Entry;
use craft\events\RegisterElementSourcesEvent;
use yii\base\Event;

Event::on(
    Entry::class,
    Entry::EVENT_REGISTER_SOURCES,
    function (RegisterElementSourcesEvent $event) {
        // Make adjustments to $event->sources here...
    }
);
```
